tests/basic/src/test/java/org/jboss/hal/testsuite/test/configuration/management/ssl/HttpManagementInterfaceTest.java

```diff
@@ -0,0 +1,779 @@
+    static final Address HTTP_INTERFACE_ADDRESS = Address.coreService(MANAGEMENT).and(MANAGEMENT_INTERFACE, HTTP_INTERFACE);
+
+    private static OnlineManagementClient client;
+    private static Operations ops;
+    private static SslOperations sslOps;
+    private static File keyStoresDirectory;
+
+    private SnapshotBackup snapshot = new SnapshotBackup();
+
+    @Inject
+    private Console console;
+
+    @Page
+    private StandaloneServerPage page;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        changeClientToNonSSL();
+        keyStoresDirectory = sslOps.getKeyStoresDirectory();
+    }
+
+    @AfterClass
+    public static void cleanUp() throws IOException {
+        FILE_NAMES_TO_BE_DELETED.stream().forEach(fileName -> FileUtils.deleteQuietly(new File(keyStoresDirectory, fileName)));
+        client.close();
+    }
+
+    @Before
+    public void backup() throws CommandFailedException, IOException {
+        client.apply(snapshot.backup());
+    }
+
+    @After
+    public void restoreBackup() throws IOException, InterruptedException, TimeoutException, CommandFailedException {
+        client.apply(snapshot.restore());
+    }
+
+    /**
+     * Testing configuration of http management interface with
+     * <ul>
+     *  <li>just one-way server authentication</li>
+     *  <li>server certificate key store to be created</li>
+     *  <li>server self-signed certificate to be generated</li>
+     * </ul>
+     */
+    @Test
+    public void enableOneWayGenerated() throws Exception {
+
+        String
+            keyStoreNameValue = Ids.build(KEY_STORE, NAME, Random.name()),
+            keyStorePasswordValue = Ids.build(KEY_STORE, PASS, Random.name()),
+            keyStorePathValue = Ids.build(KEY_STORE, PATH, Random.name()),
+            keyAliasValue = Ids.build(KEY_ALIAS, Random.name()),
+            keyManagerValue = Ids.build(KEY_MANAGER, Random.name()),
+            serverSslContextValue = Ids.build(SERVER_SSL_CONTEXT, Random.name());
+        FILE_NAMES_TO_BE_DELETED.add(keyStorePathValue);
+
+        EnableSslWizard wizard = page.enableSslWizard()
+            .tryNextToConfigurationWithExpectError(YOU_NEED_TO_SELECT_AUTHENTICATION_AS_WELL_AS_KEY_STORE_MANIPULATION_STATEGY)
+            .disableMutualAuthentication()
+            .tryNextToConfigurationWithExpectError(YOU_NEED_TO_SELECT_KEY_STORE_MANIPULATION_STATEGY)
+            .createAllResources()
+            .nextConfiguration();
+        FormFragment configForm = wizard.getConfigurationForm();
+        configForm.editTextFiringExtraChangeEvent(KEY_DN_ORGANIZATION, HAL);
+        wizard.next();
+
+        // Required fields validation test
+        configForm
+            .expectError(KEY_STORE_NAME)
+            .expectError(KEY_STORE_PASSWORD)
+            .expectError(KEY_STORE_PATH)
+            .expectError(KEY_ALIAS)
+            .expectError(KEY_MANAGER)
+            .expectError(SERVER_SSL_CONTEXT);
+
```

Comment:
I suggest refactoring the SSL configuration because lots of code is being duplicated in the `HttpManagementInterfaceTest` and `HttpsListenerTest`.